### PR TITLE
Chapter 5 carlos commments

### PR DESCRIPTION
**Pull request by TheRumle**

### CODE REVIEW

File: report/sections/Architecture/architecture.tex [hunk 1]
```diff
@@ -2,7 +2,12 @@
...
 The platform is a web application designed using a client-server architecture. Using this architecture the client communicates with a server using HTTP to send \textit{request} messages. The server waits for client requests and processes the requests when they arrive and replies to the requests with a \textit{response} message.
 
-The platform has a user interface as well as business logic to handle user requests. The platform supports role-based authentication with roles, meaning that users of varying roles have access to different functionality. The platform can run tests, to check whether a piece of code solves an exercise. Account data, test runs, and syllabus data is persisted in a database.
+The platform has a frontend component containing a user interface and logic to communicate to the backend component. 
+The backend component contains business logic to handle user requests. 
+It supports role-based authentication with roles, meaning that users of varying roles have access to different functionality.
```
Comment (chhoumann) [inline report/sections/Architecture/architecture.tex:7]:
```suggestion
It supports role-based authentication, meaning that users have access to different functionality, depending on their role. 
```
Writing roles that many times seems redundant

File: report/sections/Architecture/architecture.tex [hunk 2]
```diff
@@ -2,7 +2,12 @@
 The platform allows lecturers to create a syllabus that includes problem sets, exercises and test cases for the exercises. Additionally, students can do the exercises that the lecturer has provided.
 The platform is a web application designed using a client-server architecture. Using this architecture the client communicates with a server using HTTP to send \textit{request} messages. The server waits for client requests and processes the requests when they arrive and replies to the requests with a \textit{response} message.
 
-The platform has a user interface as well as business logic to handle user requests. The platform supports role-based authentication with roles, meaning that users of varying roles have access to different functionality. The platform can run tests, to check whether a piece of code solves an exercise. Account data, test runs, and syllabus data is persisted in a database.
+The platform has a frontend component containing a user interface and logic to communicate to the backend component.
```
Comment (chhoumann) [inline report/sections/Architecture/architecture.tex:5]: 4 of the first 7 lines start with `the platform`, seems a bit excessive

File: report/sections/Architecture/architecture.tex [hunk 3]
```diff
@@ -12,19 +17,17 @@
 	\label{fig:Architecture}
 \end{figure}
 
-Note that the platform has an API running on a separate server instance to handle code execution and test, called \textit{Test Runner}. As components are introduced in the following sections, we elaborate on the choices made for each component, in relation to this structure.
-
 \section{Backend System}
-The backend system has several responsibilities including routing, data retrieval and updates and orchestration of execution of code tests. The backend is the link between all the components in the system. It operates as a traditional server in a client-server architecture.
+The backend system has several responsibilities including routing, data retrieval and updates and orchestration of execution of code tests.
```
Comment (chhoumann) [inline report/sections/Architecture/architecture.tex:21]:
```suggestion
The backend system has several responsibilities including routing, data retrieval and mutations, and orchestration of execution of code tests. 
```
We use mutations in other places + have described it earlier in the report as being similar to creating/updating/etc. data.

File: report/sections/Architecture/architecture.tex [hunk 4]
```diff
@@ -12,19 +17,17 @@
...
+The backend system has several responsibilities including routing, data retrieval and updates and orchestration of execution of code tests. 
+The backend is the link between all the components in the system. It operates as a traditional server in a client-server architecture.
 
 \section{Frontend} \label{sec:architecture-frontend}
 The frontend is the user facing component of our web application. The frontend is designed such that its user interface and functionality is dependent on the role of the user.
 For example if a student has signed in, the student is able to see and complete exercises. In turn if a lecturer signs in, they will have access to more functionality such as creating a syllabus or exercise.
-
-Regardless of the roles of the user the frontend has the ability to send data to the backend for storage, and show the output of executed code, by sending it to the backend and showing the response.
+Regardless of the roles of the user the frontend has the ability to send data to the backend for storage, and show the output of executed code.
```
Comment (chhoumann) [inline report/sections/Architecture/architecture.tex:27]:
```suggestion
Regardless of the user's role, the frontend has the ability to send data to the backend for storage, and show the output of executed code.
```

Users only have a single role

File: report/sections/Architecture/architecture.tex (hunk 2 above)
Comment (TheRumle) [reply report/sections/Architecture/architecture.tex:5]: Det skal komme i andre rettelser, ikke dem fra carlos

**Pull request merged**